bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableConnection.java

```diff
@@ -140,4 +144,56 @@ public void clearRegionLocationCache() {
     }
     return regionInfos;
   }
+
+  @Override
+  public Hbck getHbck() throws IOException {
+    return createUnsupportedProxy(Hbck.class);
+  }
+
+  @Override
+  public Hbck getHbck(ServerName masterServer) throws IOException {
+    return createUnsupportedProxy(Hbck.class);
+  }
+
+  private <T> T createUnsupportedProxy(Class<T> cls) {
+
+    @SuppressWarnings("unchecked")
+    T proxy =
+        (T)
+            Proxy.newProxyInstance(
+                cls.getClassLoader(),
+                new Class<?>[] {cls},
+                new UnsupportedInvocationHandler("Unsupported" + cls.getName()));
+
+    return proxy;
+  }
+
+  private static class UnsupportedInvocationHandler implements InvocationHandler {
+    private final String name;
+
+    private UnsupportedInvocationHandler(String name) {
+      this.name = name;
+    }
+
+    @Override
+    public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+      if (args.length == 1
+          && method.getName().equals("equals")
+          && method.getParameterTypes()[0] == Object.class) {
+        Object arg = args[0];
+        if (arg == null) {
+          return false;
+        }
+        return proxy == arg;
+      }
+      if (args.length == 1 && method.getName().equals("hashCode")) {
+        return super.hashCode();
+      }
+      if (args.length == 0 && method.getName().equals("toString")) {
+        return name;
+      }
+
+      throw new UnsupportedOperationException(name + "." + method.getName());
```

Comment:
It's not clear to me what this is doing - can you add some docs? I would have expected this to just throw.

Reply:
I added some docs to the factory method

Reply:
I'll port it to the master version after we are done iterating on this